app/selenium_ui/conftest.py

```diff
@@ -246,19 +246,22 @@ def get_requests_by_url(requests, url_path):
     return filtered_requests
 
 
-def get_wait_browser_metrics(webdriver):
+def get_wait_browser_metrics(webdriver, expected_metrics):
     attempts = 20
-    max_wait_time = 10
-    sleep_time = float(max_wait_time / attempts)
+    sleep_time = 0.5
+    data = {}
 
-    for i in range(0, attempts):
+    for i in range(attempts):
         requests = get_performance_logs(webdriver)
         requests_bulk = get_requests_by_url(requests, 'bulk')
-        for request_id, request in requests_bulk.items():
-            if 'browser.metrics.navigation' in str(request):
-                return requests_bulk
+        data.update(requests_bulk)
+
+        if all([metric in str(data) for metric in expected_metrics]):
```

Comment:
Maybe it is worth collecting the `keys` from the data and checking if the metric is in the keys? Need to test and discuss.